ONSdigital/dp-healthcheck
# Update Go and dependencies

### What

Update Go to 1.17 and update dependencies.
This removes some vulnerabilities that make services using the current version of this package fail the audit job - this happens since v1.1.1
The problematic dependency was `github.com/gopherjs/gopherjs`

The replace statements can be removed since those packages are no longer used. 

### How to review

1. Go to a repo using this package. For example: `dp-frontend-dataset-controller`
2.  See that there are vulnerabilities with the latest version:
  - `go get -d github.com/ONSdigital/dp-healthcheck`
  - `go mod tidy`
  - `go list -json -m all | nancy sleuth`
3. Revert changes: `git checkout go.mod`
4. Check that there are no vulnerabilities when using this version:
  - `go get -d github.com/ONSdigital/dp-healthcheck@eaf2549b485f23513b813f13c67010105446d268`
  - `go mod tidy`
  - `go list -json -m all | nancy sleuth`
5. Check tests still pass


Note: you could check that the replace statements are no longer needed by running `go mod graph | grep coreos/etcd` and `go mod graph | grep gogo/protobuf` - it will not return any line

### Who can review

Anyone
